paymentsheet/src/test/java/com/stripe/android/paymentelement/confirmation/ConfirmationActivity.kt

```diff
@@ -0,0 +1,156 @@
+import com.stripe.android.googlepaylauncher.GooglePayRepository
+import com.stripe.android.networking.StripeApiRepository
+import com.stripe.android.networking.StripeRepository
+import com.stripe.android.payments.core.analytics.ErrorReporter
+import com.stripe.android.payments.core.injection.PRODUCT_USAGE
+import com.stripe.android.paymentsheet.utils.FakeUserFacingLogger
+import com.stripe.android.testing.FakeAnalyticsRequestExecutor
+import com.stripe.android.testing.FakeErrorReporter
+import com.stripe.android.testing.FakeLogger
+import dagger.Binds
+import dagger.BindsInstance
+import dagger.Component
+import dagger.Module
+import dagger.Provides
+import kotlinx.coroutines.flow.flowOf
+import javax.inject.Inject
+import javax.inject.Named
+import javax.inject.Singleton
+
+internal class ConfirmationActivity : AppCompatActivity() {
+    val viewModel: TestViewModel by viewModels {
+        TestViewModel.Factory
+    }
+
+    val confirmationHandler by lazy {
+        viewModel.confirmationHandler
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        confirmationHandler.register(activityResultCaller = this, lifecycleOwner = this)
+    }
+
+    class TestViewModel @Inject constructor(
+        confirmationHandlerFactory: ConfirmationHandler.Factory
+    ) : ViewModel() {
+        val confirmationHandler = confirmationHandlerFactory.create(viewModelScope)
+
+        object Factory : ViewModelProvider.Factory {
+            override fun <T : ViewModel> create(modelClass: Class<T>, extras: CreationExtras): T {
+                val component = DaggerConfirmationTestComponent.builder()
+                    .application(extras.requireApplication())
+                    .allowsManualConfirmation(allowsManualConfirmation = false)
+                    .statusBarColor { null }
+                    .savedStateHandle(extras.createSavedStateHandle())
+                    .build()
+
+                @Suppress("UNCHECKED_CAST")
+                return component.viewModel as T
+            }
+        }
+    }
+}
+
+@Component(
+    modules = [
+        ConfirmationModule::class,
+        CoroutineContextModule::class,
+        ConfirmationTestModule::class,
+    ]
+)
+@Singleton
+internal interface ConfirmationTestComponent {
+    val viewModel: ConfirmationActivity.TestViewModel
+
+    @Component.Builder
+    interface Builder {
+        @BindsInstance
+        fun application(application: Application): Builder
+
+        @BindsInstance
+        fun savedStateHandle(savedStateHandle: SavedStateHandle): Builder
+
+        @BindsInstance
+        fun statusBarColor(@Named(STATUS_BAR_COLOR_PROVIDER) statusBarColor: () -> Int?): Builder
+
+        @BindsInstance
+        fun allowsManualConfirmation(@Named(ALLOWS_MANUAL_CONFIRMATION) allowsManualConfirmation: Boolean): Builder
+
+        fun build(): ConfirmationTestComponent
+    }
+}
```

Comment:
We can replace this `Component` with multiple variants once we have injection:
- `BaseTestComponent` (only intent confirmation)
- `PaymentElementTestComponent` (intent, GPay, Bacs, EPMs)
- `ExtendedPaymentElementTestComponent` (intent, GPay, Bacs, EPMs, CVC, SEPA)